server/src/main/java/io/crate/execution/engine/aggregation/impl/CmpByAggregation.java

```diff
@@ -105,6 +154,35 @@ public CmpByAggregation(int cmpResult, Signature signature, BoundSignature bound
         }
     }
 
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver referenceResolver,
+                                                       List<Reference> aggregationReferences,
+                                                       DocTableInfo table,
+                                                       List<Literal<?>> optionalParams) {
+        Reference searchRef = aggregationReferences.get(1);
```

Comment:
Could you please add a comment why this is `1` and line 174 is `0`?

Reply:
Doesn't the name `searchRef` and `resultExpression` give enough indication that it refers to the search and result parameter of the function?